docs/examples/foo_detector.py

```diff
@@ -3,22 +3,19 @@
 
 from bluesky.protocols import HasHints, Hints
 
-from ophyd_async.core import DirectoryProvider
-from ophyd_async.core.async_status import AsyncStatus
-from ophyd_async.core.detector import DetectorControl, DetectorTrigger, StandardDetector
-from ophyd_async.epics.areadetector.drivers.ad_base import (
-    ADBase,
-    ADBaseShapeProvider,
-    start_acquiring_driver_and_ensure_status,
+from ophyd_async.core import (
+    AsyncStatus,
+    DetectorControl,
+    DetectorTrigger,
+    DirectoryProvider,
+    StandardDetector,
 )
-from ophyd_async.epics.areadetector.utils import ImageMode, ad_rw, stop_busy_record
-from ophyd_async.epics.areadetector.writers.hdf_writer import HDFWriter
-from ophyd_async.epics.areadetector.writers.nd_file_hdf import NDFileHDF
+from ophyd_async.epics import adcore
 
 
-class FooDriver(ADBase):
+class FooDriver(adcore.ADBase):
     def __init__(self, prefix: str, name: str = "") -> None:
-        self.trigger_mode = ad_rw(str, prefix + "TriggerMode")
+        self.trigger_mode = adcore.ad_rw(str, prefix + "TriggerMode")
```

Comment:
#274 deleted `adcore.ad_rw` and replaced with `epics_signal_rw_rbv`. Please can you do this one two? And see if the docs build is giving an error that we are ignoring...

Reply:
Fixed this, and also the outdated DirectoryProvider